src/jit/codegencommon.cpp

```diff
@@ -639,6 +639,19 @@ regMaskTP Compiler::compNoGCHelperCallKillSet(CorInfoHelpFunc helper)
 
     switch (helper)
     {
+        case CORINFO_HELP_ASSIGN_BYREF:
+#if defined(_TARGET_X86_)
+            // This helper only trashes ECX.
+            return RBM_ECX;
+#elif defined(_TARGET_AMD64_)
+            // This uses and defs rdi and rci.
```

Comment:
Typo: rci -> rsi
Also, you may want to just be quite explicit that "This uses and defines rdi and rsi, so they remain as live GC refs or byrefs, and are not killed." or something like that.